File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/OnHurtMessage.ts
import { IMessage } from "../../../../Core/ObserverMessageKit";
import DamageInfo from "../DataDefine/DamageInfo";
import AbstractCreature from "../Unit/AbstractCreature";

export class OnHurtMessage implements IMessage{
    public static readonly Id = "OnHurtMessage";
    public constructor(public Damage:DamageInfo,public target:AbstractCreature){
        
    }    
}

File: Unity/Assets/Scripts/ModTheSpire/Event/Ids/OnHurtMessage.cs
using ModTheSpire.DataDefine;
using ModTheSpire.Unit;

namespace ModTheSpire.Event.Ids
{
    ///受到伤害时
    public class OnHurtMessage : IEventHandle
    {
        public DamageInfo DamageInfo;
        public AbstractCreature Target;

        public OnHurtMessage(DamageInfo damageInfo, AbstractCreature target)
        {
            DamageInfo = damageInfo;
            Target = target;
        }
    }
}